lib/game_team.rb

```diff
@@ -116,6 +120,31 @@ def self.worst_offense
     total_goals_per_team= Hash[grouped_team.keys.zip(team_averaged_goals)]
     total_goals_per_team.key(total_goals_per_team.values.min)
   end
+
+  def self.most_goals_scored(team_id)
+  total_game_teams_per_team_id = find_by_team(team_id)
+  results = {}
+  total_game_teams_per_team_id.each do |game_team|
+    results[game_team.game_id] ||= {"team_id"=>0, "goals"=>0}
+    results[game_team.game_id]["team_id"] = game_team.team_id
+    results[game_team.game_id]["goals"] = game_team.goals
+  end
+  max_goals = results.max_by{|key,value| value["goals"]}
+  return max_goals[1]["goals"]
+  end
+
+  def self.least_goals_scored(team_id)
+    total_game_teams_per_team_id = find_by_team(team_id)
+    results = {}
+    total_game_teams_per_team_id.each do |game_team|
+      results[game_team.game_id] ||= {"team_id"=>0, "goals"=>0}
+      results[game_team.game_id]["team_id"] = game_team.team_id
+      results[game_team.game_id]["goals"] = game_team.goals
+      end
+    min_goals = results.min_by{|key,value| value["goals"]}
+    return min_goals[1]["goals"]
+  end
```

Comment:
These two methods can be modulized with hash_of_hashes!

Reply:
agreed!